packages/core/src/model.ts

```diff
@@ -426,6 +427,18 @@ export class Model implements Storage {
     return result
   }
 
+  getClassMixins (cls: Ref<Class<Obj>>): Ref<Mixin<Doc>>[] {
+    const result = [] as Ref<Mixin<Doc>>[]
+    for (const doc of this.objects.values()) {
```

Comment:
You have missed a mixins extends mixins.

Please refer to PR #413 it has a method to get all extends for a class.